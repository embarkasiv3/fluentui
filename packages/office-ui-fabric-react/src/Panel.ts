export * from './packages/react-fundamentals/Panel';
