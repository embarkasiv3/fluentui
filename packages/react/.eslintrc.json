{
  // react-internal has some overrides we need to mirror
  "extends": ["../react-internal/.eslintrc"],
  "root": true
}
