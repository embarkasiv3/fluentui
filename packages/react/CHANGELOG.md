# Change Log - @fluentui/react

This log was last generated on Thu, 19 Mar 2020 17:52:34 GMT and should not be manually modified.
