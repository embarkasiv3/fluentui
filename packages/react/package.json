{
  "name": "@fluentui/react",
  "version": "8.0.0",
  "description": "Reusable React components for building web experiences.",
  "main": "lib-commonjs/index.js",
  "module": "lib/index.js",
  "typings": "lib/index.d.ts",
  "sideEffects": [
    "lib/version.js"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/microsoft/fluentui"
  },
  "license": "MIT",
  "scripts": {
    "build": "just-scripts build",
    "bundle": "just-scripts bundle",
    "clean": "just-scripts clean",
    "code-style": "just-scripts code-style",
    "codepen": "node ../../scripts/local-codepen.js",
    "just": "just-scripts",
    "lint": "just-scripts lint",
    "start": "just-scripts dev:storybook",
    "start:legacy": "yarn workspace @fluentui/public-docsite-resources start",
    "start-test": "just-scripts jest-watch",
    "test": "just-scripts test",
    "update-snapshots": "just-scripts jest -u",
    "mf": "just-scripts mf"
  },
  "devDependencies": {
    "@fluentui/eslint-plugin": "^1.0.0",
    "@types/chalk": "^2.0.0",
    "@types/enzyme": "3.10.3",
    "@types/enzyme-adapter-react-16": "1.0.3",
    "@types/glob": "^7.1.1",
    "@types/node": "^10.3.2",
    "@types/react": "16.9.42",
    "@types/react-dom": "16.9.10",
    "@types/react-test-renderer": "^16.0.0",
    "@types/webpack-env": "1.16.0",
    "@fluentui/scripts": "^1.0.0",
    "@fluentui/example-data": "^8.0.0",
    "@fluentui/jest-serializer-merge-styles": "^8.0.0",
    "@fluentui/test-utilities": "^8.0.0",
    "@fluentui/webpack-utilities": "^8.0.0",
    "@fluentui/react-conformance": "^0.2.1",
    "chalk": "^2.1.0",
    "enzyme": "~3.10.0",
    "enzyme-adapter-react-16": "^1.15.0",
    "glob": "^7.1.2",
    "jest-snapshot": "~24.9.0",
    "office-ui-fabric-core": "^11.0.0",
    "react": "16.8.6",
    "react-app-polyfill": "~1.0.1",
    "react-dom": "16.8.6",
    "react-test-renderer": "^16.3.0"
  },
  "dependencies": {
    "@fluentui/date-time-utilities": "^8.0.0",
    "@fluentui/foundation-legacy": "^8.0.0",
    "@fluentui/font-icons-mdl2": "^8.0.0",
    "@fluentui/merge-styles": "^8.0.0",
    "@fluentui/react-focus": "^8.0.0",
    "@fluentui/react-hooks": "^8.0.0",
    "@fluentui/react-shared-contexts": "^1.0.0-beta.9",
    "@fluentui/react-window-provider": "^2.0.0",
    "@fluentui/set-version": "^8.0.0",
    "@fluentui/style-utilities": "^8.0.0",
    "@fluentui/theme": "^2.0.0",
    "@fluentui/utilities": "^8.0.0",
    "@microsoft/load-themed-styles": "^1.10.26",
    "tslib": "^1.10.0"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 <18.0.0",
    "@types/react-dom": ">=16.8.0 <18.0.0",
    "react": ">=16.8.0 <18.0.0",
    "react-dom": ">=16.8.0 <18.0.0"
  }
}
