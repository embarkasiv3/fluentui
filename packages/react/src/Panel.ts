export * from '@fluentui/react-internal/lib/Panel';
