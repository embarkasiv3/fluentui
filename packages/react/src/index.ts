import './version';
export * from '@fluentui/react-internal';
export * from './Breadcrumb';
export * from './Checkbox';
export * from './ComboBox';
export * from './DetailsList';
export * from './DocumentCard';
export * from './Dropdown';
export * from './GroupedList';
export * from './Link';
export * from './Pivot';
export * from './ShimmeredDetailsList';
export * from './Slider';
export * from './Toggle';
