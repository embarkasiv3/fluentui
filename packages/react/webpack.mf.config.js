const path = require('path');
const fs = require('fs');
const webpack = require('webpack');
const getResolveAlias = require('@fluentui/scripts/webpack/getResolveAlias');
const { createServeConfig } = require('@fluentui/scripts/webpack/webpack-resources');
const { webpackMerge } = require('just-scripts');

const shared = {
  react: { singleton: true },
  'react-dom': { singleton: true },
};

const rootComponents = fs
  .readdirSync(path.join(__dirname, 'src'))
  .filter(filename => filename[0].toUpperCase() === filename[0])
  .map(filename => filename.replace(/\.tsx?/, ''));
const rootComponentsExposes = {};
for (const component of rootComponents) {
  rootComponentsExposes[`./lib/${component}`] = `@fluentui/react/src/${component}`;
}

const myConfig = {
  output: {},
  entry: {},
  mode: 'development',
  devtool: 'cheap-module-source-map',
  optimization: {
    concatenateModules: false,
  },
  module: {
    rules: [],
  },
  plugins: [
    new webpack.container.ModuleFederationPlugin({
      name: 'fluentuiReact',
      filename: 'remoteEntry.js',
      exposes: {
        ...rootComponentsExposes,
        './lib/compat/Button': '@fluentui/react/src/compat/Button',
        '.': '@fluentui/react/src/index',
        './lib/index': '@fluentui/react/src/index',
      },
      shared,
    }),
  ],
  resolve: {
    alias: getResolveAlias(),
  },
  devServer: {
    port: 2345,
    contentBase: 'dist',
  },
};

const metaConfig = webpackMerge.merge(createServeConfig(), myConfig);

module.exports = metaConfig;
