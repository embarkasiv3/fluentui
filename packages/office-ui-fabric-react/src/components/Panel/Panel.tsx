/* tslint:disable:no-unused-variable */
import * as React from 'react';
/* tslint:enable:no-unused-variable */

import {
  BaseComponent,
  css,
  getId,
  getRTL
} from '../../Utilities';
import { FocusTrapZone } from '../FocusTrapZone/index';
import { IPanelProps, PanelType } from './Panel.Props';
import { Layer } from '../Layer/Layer';
import { Overlay } from '../../Overlay';
import { Popup } from '../../Popup';
import { IconButton } from '../../Button';
import styles from './Panel.scss';

export interface IPanelState {
  isOpen?: boolean;
  isAnimatingOpen?: boolean;
  isAnimatingClose?: boolean;
  id?: string;
}

// Animation class constants.
const FADE_IN_200 = 'ms-u-fadeIn200';
const FADE_OUT_200 = 'ms-u-fadeOut200';
const SLIDE_LEFT_IN_40 = 'ms-u-slideLeftIn40';
const SLIDE_LEFT_OUT_40 = 'ms-u-slideLeftOut40';
const SLIDE_RIGHT_IN_40 = 'ms-u-slideRightIn40';
const SLIDE_RIGHT_OUT_40 = 'ms-u-slideRightOut40';

export class Panel extends BaseComponent<IPanelProps, IPanelState> {

  public static defaultProps: IPanelProps = {
    isOpen: false,
    isBlocking: true,
    hasCloseButton: true,
    type: PanelType.smallFixedFar,
  };

  constructor(props: IPanelProps) {
    super(props);

    this._onPanelClick = this._onPanelClick.bind(this);
    this._onPanelRef = this._onPanelRef.bind(this);

    this.state = {
      isOpen: !!props.isOpen,
      isAnimatingOpen: props.isOpen,
      isAnimatingClose: false,
      id: getId('Panel')
    };
  }

  public componentDidMount() {
    if (this.state.isOpen) {
      this._async.setTimeout(() => {
        this.setState({
          isAnimatingOpen: false
        });
      }, 2000);
    }
  }

  public componentWillReceiveProps(newProps: IPanelProps) {
    if (newProps.isOpen !== this.state.isOpen) {
      this.setState({
        isOpen: true,
        isAnimatingOpen: newProps.isOpen ? true : false,
        isAnimatingClose: newProps.isOpen ? false : true
      });
    }
  }

  public render() {
    let {
      children,
      className = '',
      closeButtonAriaLabel,
      elementToFocusOnDismiss,
      firstFocusableSelector,
      forceFocusInsideTrap,
      hasCloseButton,
      headerClassName = '',
      headerText,
      ignoreExternalFocusing,
      isBlocking,
      isLightDismiss,
      layerProps,
      type
    } = this.props;
    let { isOpen, isAnimatingOpen, isAnimatingClose, id } = this.state;
    let isLeft = type === PanelType.smallFixedNear ? true : false;
    let isRTL = getRTL();
    let isOnRightSide = isRTL ? isLeft : !isLeft;
    const headerTextId = id + '-headerText';

    let pendingCommandBarContent = '';

    if (!isOpen) {
      return null;
    }

    let header;
    if (headerText) {
      header = <p className={ css('ms-Panel-headerText', styles.headerText, headerClassName) } id={ headerTextId }>{ headerText }</p>;
    }

    let closeButton;
    if (hasCloseButton) {
      closeButton = (
        <IconButton
          className={ css('ms-Panel-closeButton ms-PanelAction-close', styles.closeButton) }
          onClick={ this._onPanelClick }
          aria-label={ closeButtonAriaLabel }
          data-is-visible={ true } icon='Cancel'
        />
      );
    }

    let overlay;
    if (isBlocking) {
      overlay = (
        <Overlay
          className={ css(
            styles.overlay,
            {
              [FADE_IN_200]: isAnimatingOpen,
              [FADE_OUT_200]: isAnimatingClose
            }) }
          isDarkThemed={ false }
          onClick={ isLightDismiss ? this._onPanelClick : null }
        />
      );
    }

    return (
      <Layer { ...layerProps }>
        <Popup
          role='dialog'
          ariaLabelledBy={ headerText && headerTextId }
          onDismiss={ this.props.onDismiss }>
          <div
            ref={ this._onPanelRef }
            className={
              css('ms-Panel', styles.root, className, {
                // because the RTL animations are not being used, we need to set a class
                ['is-open ' + styles.rootIsOpen]: isOpen,
                ['ms-Panel--smFluid ' + styles.rootIsSmallFluid]: type === PanelType.smallFluid,
                ['ms-Panel--smLeft ' + styles.rootIsSmallLeft]: type === PanelType.smallFixedNear,
                ['ms-Panel--sm ' + styles.rootIsSmall]: type === PanelType.smallFixedFar,
                ['ms-Panel--md ' + styles.rootIsMedium]: type === PanelType.medium,
                ['ms-Panel--lg ' + styles.rootIsLarge]: type === PanelType.large || type === PanelType.largeFixed,
                ['ms-Panel--fixed ' + styles.rootIsFixed]: type === PanelType.largeFixed,
                ['ms-Panel--xl ' + styles.rootIsXLarge]: type === PanelType.extraLarge,
                ['ms-Panel--hasCloseButton ' + styles.rootHasCloseButton]: hasCloseButton
              })
            }
          >
            { overlay }
            <FocusTrapZone
              className={ css(
                'ms-Panel-main',
                styles.main,
                {
                  [SLIDE_RIGHT_IN_40]: isAnimatingOpen && !isOnRightSide,
                  [SLIDE_LEFT_IN_40]: isAnimatingOpen && isOnRightSide,
                  [SLIDE_LEFT_OUT_40]: isAnimatingClose && !isOnRightSide,
                  [SLIDE_RIGHT_OUT_40]: isAnimatingClose && isOnRightSide
                }
              ) }
              elementToFocusOnDismiss={ elementToFocusOnDismiss }
              isClickableOutsideFocusTrap={ isLightDismiss }
              ignoreExternalFocusing={ ignoreExternalFocusing }
              forceFocusInsideTrap={ forceFocusInsideTrap }
              firstFocusableSelector={ firstFocusableSelector }
            >
              <div className={ css('ms-Panel-commands') } data-is-visible={ true } >
                { pendingCommandBarContent }
                { closeButton }
              </div>
              <div className={ css('ms-Panel-contentInner', styles.contentInner) }>
                { header }
                <div className={ css('ms-Panel-content') }>
                  { children }
                </div>
              </div>
            </FocusTrapZone>
          </div>
        </Popup>
      </Layer>
    );
  }

  public componentDidUpdate(prevProps, prevState) {
    if (
      prevState.isAnimatingClose === false &&
      this.state.isAnimatingClose === true &&
      this.props.onDismiss) {
      this.props.onDismiss();
    }
  }

  public dismiss() {
    if (this.state.isOpen) {
      this.setState({
        isAnimatingOpen: false,
        isAnimatingClose: true
      });
    }
  }

  private _onPanelClick() {
    this.dismiss();
  }

  private _onPanelRef(ref: HTMLDivElement) {
    if (ref) {
      this._events.on(ref, 'animationend', this._onAnimationEnd);
    } else {
      this._events.off();
    }
  }

  private _onAnimationEnd(ev: AnimationEvent) {
    if (ev.animationName.indexOf('In') > -1) {
      this.setState({
        isOpen: true,
        isAnimatingOpen: false
      });
    }
    if (ev.animationName.indexOf('Out') > -1) {
      this.setState({
        isOpen: false,
        isAnimatingClose: false
      });

      if (this.props.onDismissed) {
        this.props.onDismissed();
      }
    }
  }
}
