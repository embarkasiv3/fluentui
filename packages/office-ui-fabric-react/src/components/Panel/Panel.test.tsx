import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { PanelBase } from './Panel.base';
import { Panel } from './Panel';
let div: HTMLElement;

const ReactDOM = require('react-dom');

describe('Panel', () => {
  afterEach(() => {
    jest.useRealTimers();
  });

  it('renders Panel correctly', () => {
    // Mock createPortal to capture its component hierarchy in snapshot output.
    ReactDOM.createPortal = jest.fn(element => {
      return element;
    });

    const component = renderer.create(
      <Panel isOpen={true} headerText="Test Panel">
        <span>Content goes here</span>
      </Panel>
    );

    expect(component.toJSON()).toMatchSnapshot();

    ReactDOM.createPortal.mockClear();
  });

  describe('open', () => {
    beforeEach(() => {
      div = document.createElement('div');
    });

    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div);
    });

    it('fires the correct events when opening', () => {
      let openedCalled = false;
      let openCalled = false;
      const setOpenTrue = (): void => {
        openCalled = true;
      };
      const setOpenedTrue = (): void => {
        openedCalled = true;
      };

      jest.useFakeTimers();

      const panel: PanelBase = ReactDOM.render(<PanelBase isOpen={false} onOpen={setOpenTrue} onOpened={setOpenedTrue} />, div) as any;

      panel.open();

      expect(openCalled).toEqual(true);
      expect(openedCalled).toEqual(false);

      jest.runOnlyPendingTimers();

      expect(openedCalled).toEqual(true);
    });
  });

  describe('onClose', () => {
    beforeEach(() => {
      div = document.createElement('div');
    });

    afterEach(() => {
      ReactDOM.unmountComponentAtNode(div);
    });

    it('fires the correct events when closing', () => {
      let dismissedCalled = false;
      let dismissCalled = false;
      const setDismissTrue = (): void => {
        dismissCalled = true;
      };
      const setDismissedTrue = (): void => {
        dismissedCalled = true;
      };

      jest.useFakeTimers();

      const panel: PanelBase = ReactDOM.render(
        <PanelBase isOpen={true} onDismiss={setDismissTrue} onDismissed={setDismissedTrue} />,
        div
      ) as any;

      panel.dismiss();

      expect(dismissCalled).toEqual(true);
      expect(dismissedCalled).toEqual(false);

      jest.runOnlyPendingTimers();

      expect(dismissedCalled).toEqual(true);
    });
  });
});
