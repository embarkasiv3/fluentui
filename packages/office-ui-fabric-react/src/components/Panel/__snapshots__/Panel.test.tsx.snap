// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel renders Panel correctly 1`] = `
<div
  aria-describedby={undefined}
  aria-label={undefined}
  aria-labelledby="Panel0-headerText"
  className=""
  onKeyDown={[Function]}
  role="dialog"
  style={
    Object {
      "overflowY": undefined,
    }
  }
>
  <div
    className="ms-Panel is-open undefined ms-Panel--sm undefined ms-Panel--hasCloseButton undefined"
    type={1}
  >
    <div
      className=
          ms-Overlay
          {
            animation-duration: 0.267s;
            animation-fill-mode: both;
            animation-name: keyframes from{opacity:0;}to{opacity:1;};
            animation-timing-function: cubic-bezier(.1,.25,.75,.9);
            background-color: rgba(255,255,255,.4);
            bottom: 0px;
            left: 0px;
            position: absolute;
            right: 0px;
            top: 0px;
          }
          @media screen and (-ms-high-contrast: active){& {
            border: 1px solid WindowText;
          }
      onClick={undefined}
    />
    <div
      aria-labelledby={undefined}
      className=
          ms-Panel-main
          {
            animation-duration: 0.367s;
            animation-fill-mode: both;
            animation-name: keyframes from{opacity:0;}to{opacity:1;} keyframes from{transform:translate3d(40px,0,0);}to{transform:translate3d(0,0,0);};
            animation-timing-function: cubic-bezier(.1,.9,.2,1);
          }
      onFocusCapture={[Function]}
      onKeyDown={[Function]}
      style={Object {}}
    >
      <div
        className="ms-Panel-scrollableContent"
      >
        <div
          className="ms-Panel-commands"
          data-is-visible={true}
        >
          <div
            className="ms-Panel-navigation"
          >
            <button
              aria-describedby={undefined}
              aria-disabled={undefined}
              aria-label={undefined}
              aria-labelledby={undefined}
              aria-pressed={undefined}
              className=
                  ms-Button
                  ms-Button--icon
                  ms-Panel-closeButton
                  ms-PanelAction-close
                  {
                    -moz-osx-font-smoothing: grayscale;
                    -webkit-font-smoothing: antialiased;
                    background-color: transparent;
                    border-radius: 0px;
                    border: 1px solid transparent;
                    box-sizing: border-box;
                    color: #666666;
                    cursor: pointer;
                    display: inline-block;
                    font-family: 'Segoe UI', 'Segoe UI Web (West European)', 'Segoe UI', -apple-system, BlinkMacSystemFont, 'Roboto', 'Helvetica Neue', sans-serif;
                    font-size: 20px;
                    font-weight: 400;
                    height: auto;
                    outline: transparent;
                    padding-bottom: 0;
                    padding-left: 4px;
                    padding-right: 4px;
                    padding-top: 0;
                    position: relative;
                    text-align: center;
                    text-decoration: none;
                    user-select: none;
                    vertical-align: top;
                    width: 44px;
                  }
                  &::-moz-focus-inner {
                    border: 0;
                  }
                  .ms-Fabric--isFocusVisible &:focus:after {
                    border: 1px solid #ffffff;
                    bottom: 0px;
                    content: "";
                    left: 0px;
                    outline: 1px solid #666666;
                    position: absolute;
                    right: 0px;
                    top: 0px;
                    z-index: 1;
                  }
                  @media screen and (-ms-high-contrast: active){.ms-Fabric--isFocusVisible &:focus:after {
                    border: none;
                    bottom: -2px;
                    left: -2px;
                    outline-color: ButtonText;
                    right: -2px;
                    top: -2px;
                  }
                  &:hover {
                    color: #333333;
                  }
                  @media screen and (-ms-high-contrast: active){&:hover {
                    border-color: Highlight;
                    color: Highlight;
                  }
                  &:active {
                    color: #0078d4;
                  }
              data-is-focusable={true}
              data-is-visible={true}
              disabled={undefined}
              onClick={[Function]}
              onKeyDown={[Function]}
              onKeyPress={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseUp={[Function]}
              type="button"
            >
              <div
                className=
                    ms-Button-flexContainer
                    {
                      align-items: center;
                      display: flex;
                      flex-wrap: nowrap;
                      height: 100%;
                      justify-content: center;
                    }
              >
                <i
                  aria-hidden={true}
                  className=
                      ms-Button-icon
                      {
                        -moz-osx-font-smoothing: grayscale;
                        -webkit-font-smoothing: antialiased;
                        display: inline-block;
                        flex-shrink: 0;
                        font-family: "FabricMDL2Icons";
                        font-size: 16px;
                        font-style: normal;
                        font-weight: normal;
                        height: 16px;
                        line-height: 16px;
                        margin-bottom: 0;
                        margin-left: 4px;
                        margin-right: 4px;
                        margin-top: 0;
                        speak: none;
                        text-align: center;
                        vertical-align: middle;
                      }
                  data-icon-name="Cancel"
                  role="presentation"
                >
                  îœ‘
                </i>
              </div>
            </button>
          </div>
        </div>
        <div
          className="ms-Panel-contentInner"
        >
          <div
            className="ms-Panel-header"
          >
            <p
              aria-level={2}
              className="ms-Panel-headerText"
              id="Panel0-headerText"
              role="heading"
            >
              Test Panel
            </p>
          </div>
          <div
            className="ms-Panel-content"
            data-is-scrollable={true}
          >
            <span>
              Content goes here
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
